### PR TITLE
Add and use ClientOptions

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add and use ClientOptions

## Links to any relevant issues

#42 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Running examples and tests

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by thibault-martinez — DISMISSED** [review]

File: bindings/python/iota_sdk/types/client_options.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    nodeSyncInterval: Optional[Duration] = None
+    quorum: Optional[bool] = None
+    minQuorumSize: Optional[int] = None
+    quorumThreshold: Optional[int] = None
+    networkInfo: Optional[NetworkInfo] = None
+    brokerOptions: Optional[MqttBrokerOptions] = None
+    apiTimeout: Optional[Duration] = None
+    remotePowTimeout: Optional[Duration] = None
+    powWorkerCount: Optional[int] = None
+    localPow: Optional[bool] = None
```
Comment (Alex6323) [inline bindings/python/iota_sdk/types/client_options.py]: Those fields should probably be documented now right away?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/client_options.py]: Added

Comment (Alex6323) [reply bindings/python/iota_sdk/types/client_options.py]: thanks ... maybe have them in the same order as the fields? Makes it a bit easier to see whether everything is documented.

**Review by Alex6323 — DISMISSED** [review]

**Review by thibault-martinez — DISMISSED** [review]

File: bindings/python/iota_sdk/types/client_options.py (hunk 1 above)
Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/client_options.py]: Updated

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

File: bindings/python/iota_sdk/types/client_options.py (hunk 1 above)
Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/client_options.py]: Closing for you @Alex6323 but please remember to close your comments

**Pull request merged**